underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java

```diff
@@ -572,6 +573,32 @@ protected String getRootKey() {
     }
   }
 
+  @Override
+  public boolean isDirectory(String path) throws IOException {
```

Comment:
why can't we do the logic in `ObjectUnderFileSystem` so all object stores are fixed?

Reply:
done